src/main/java/ContactList.java

```diff
@@ -95,11 +95,11 @@ private int find(Friend friend) {
      *
      * @param friend Friend object to find the index of within the collection.
      * @return Index of the Friend object.
-     * @throws NoSuchElementException If no equal Friend object exists within the collection, throw an exception.
+     * @throws NoSuchElementException Throw if no equal Friend object exists within the collection, throw an exception.
      */
     public int indexOf(Friend friend) {
         if (!contains(friend)) {
-            throw new NoSuchElementException();
+            throw new NoSuchElementException("Friend does not exist within ContactList.");
```

Comment:
It's typical to include as much information as reasonably possible in your exception messages. They only occur on failures, so spending some extra time generating a good message is cheap enough. I'd probably just do:
```java
throw new NoSuchElementException(Objects.toString(friend));
```
The `NoSuchElementException` already tells you that something does not exist, and the stack trace will tell you it's a `ContactList`. The value is the only thing missing.

Reply:
Switched the index out of bouts exception to also have the index in the mesage